### PR TITLE
Add Object#lexicon?

### PR DESCRIPTION
**Pull request by komainu8**

### CODE REVIEW

**Comment by komainu8** [comment]
@kenhys Could you review this PR, please?

**Comment by komainu8** [comment]
@kou Could you review this PR, please?

File: test/test-index-column.rb [hunk 1]
```diff
@@ -63,6 +63,10 @@ def test_data?
         not @index.data?
       end
     end
+
+    def test_lexicon?
+      assert_not_predicate(@index, :lexicon?)
```
Comment (kou) [inline test/test-index-column.rb]: Could you use power-assert?

```ruby
assert do
  not @index.lexicon?
end
```

It shows receiver on failure.

**Comment by komainu8** [comment]
@kou Thank you for your comments.
I've fixed them.

File: test/test-index-column.rb [hunk 2]
```diff
@@ -63,6 +63,15 @@ def test_data?
...
     end
+
+    def test_lexicon?
+      assert do
+        @terms.lexicon?
+      end
+      assert do
+        not @index.lexicon?
+      end
+    end
```
Comment (kou) [inline test/test-index-column.rb]: Could you split these assertions? One test should check one thing. If the first check is failed, the second check isn't ran. It reduces information to debug.

```ruby
sub_test_case("#lexicon?") do
  def test_true
     assert do
      @terms.lexicon?
    end
  end

  def test_false
    assert do
      not @index.lexicon?
    end
  end
end
```

**Comment by komainu8** [comment]
@kou Thank you for your comment.
I've fixed it.

**Pull request merged**